interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -195,20 +201,65 @@ func (pc *promptConstraints) equals(other *promptConstraints) bool {
 	return true
 }
 
-// subtractPermissions removes all of the given permissions from the list of
-// permissions in the constraints.
-func (pc *promptConstraints) subtractPermissions(permissions []string) (modified bool) {
-	newPermissions := make([]string, 0, len(pc.remainingPermissions))
+// applyRuleConstraints modifies the prompt constraints, removing any remaining
+// permissions which are matched by the given rule constraints.
+//
+// Returns whether the prompt constraints were modified, whether the prompt
+// requires a response (either because all permissions were allowed or at least
+// one permission was denied), and the list of any permissions which were
+// denied. If an error occurs, it is returned, and the other parameters can be
+// ignored.
+//
+// If the path pattern does not match the prompt path, or the rule constraints
+// do not include any of the remaining prompt permissions, then modified is
+// false, and no changes are made to the prompt constraints.
+func (pc *promptConstraints) applyRuleConstraints(constraints *prompting.RuleConstraints) (modified, respond bool, denied []string, err error) {
```

Comment:
maybe coveredByRule is a better concept than modified here

Reply:
To me, "covered" seems to imply the prompt constraints were entirely satisfied by the rule constraints, but what `modified` indicates here is that at least one permission was satisfied. `respond` indicates that the prompt constraints were satisfied by the rule constraints, either because at least one permission was denied or all permissions were allowed.

How about `affectedByRule` instead of `coveredByRule`, to replace `modified`?

Reply:
affectedByRule is also ok